newrelic-agent/src/main/java/com/newrelic/agent/service/analytics/SpanEventFactory.java

```diff
@@ -399,6 +406,14 @@ public SpanEventFactory setExternalParameterAttributes(ExternalParameters parame
             } else {
                 setAddress(datastoreParameters.getHost(), datastoreParameters.getPathOrId());
             }
+        } else if (parameters instanceof MessageProduceParameters) {
+            MessageProduceParameters messageProduceParameters = (MessageProduceParameters) parameters;
+            setCategory(SpanCategory.generic);
+            setCloudResourceId(messageProduceParameters.getCloudResourceId());
```

Comment:
I found a bug where the `cloud.resource_id` attribute was never actually getting added to external spans for message producers & consumers when calling `cloudResourceId(arn)` on the `MessageProduceParameters`/`MessageConsumeParameters` builders with a valid ARN. I pushed up a fix and tests for that.

Technically, I think the same problem would exist if reporting an external with a `HttpParameters` type. This doesn’t look like it will be a problem for any of our AWS instrumentation as S3 is the only one that reports externals with `HttpParameters` as I don’t think we can actually build an ARN for S3 anyways. This might be a problem in the future though with other AWS SDKs or other vendors, so maybe we should also just `setCloudResourceId` on  `HttpParameters` as well?